framework/src/modules/dpos_v2/auxiliary.ts

```diff
@@ -0,0 +1,64 @@
+import { BlockAfterExecuteContext } from '../../node/state_machine/types';
+import { MODULE_ID_DPOS, STORE_PREFIX_DELEGATE } from './constants';
+import { delegateStoreSchema } from './schemas';
+import { DelegateAccount, ModuleConfig, ValidatorsAPI } from './types';
+
+export const updateDelegateProductivity = async ({
+	validatorsAPI,
+	previousTimestamp,
+	context,
+	moduleConfig,
+}: {
+	validatorsAPI: ValidatorsAPI;
+	previousTimestamp: number;
+	context: BlockAfterExecuteContext;
+	moduleConfig: ModuleConfig;
+}) => {
+	const { header, getAPIContext, getStore } = context;
+
+	const newHeight = header.height;
+	const apiContext = getAPIContext();
+	const missedBlocks = await validatorsAPI.getGeneratorsBetweenTimestamps(
```

Comment:
Does it return missed blocks or the list of generators with number of slots

Reply:
It returns an object where object keys are generator addresses and object values are amount of blocks that should be forged by the corresponding generator address between two timestamps. In this context if we get generators between previous timestamp - which is the last forged moment - and current timestamp the result we get is equal to generators that missed blocks with also amount of blocks missed.